### ISSUE TITLE
--tls generates error, drops events

### ISSUE THREAD

**Issue by tempire** [riemann/riemann-tools#288]
## Issue

The --tls switch is used:

riemann-health --host example.com --port 5554 --tcp --tls true

No events are sent to the server.

`Dropped 113 events due to no implicit conversion of nil into String
`
### What's your riemann-tools version?

riemann-tools (1.10.0)

### What platform are you on?

Same result on macos, alpine, and ami2 linux. Works fine when not sending to tls-enabled server. 

Events work fine using tls in nodejs.

**Comment by tempire** [comment]
Yes, without ca-cert, cert, and key. I want secure communication, not authentication. The server has a letsencrypt-issued cert.

**Comment by smortex** [comment]
> I want secure communication, not authentication

Hum, is this supported by riemann?  The server seems to assert presence of cert, key and ca-cert to start a TLS server:
https://github.com/riemann/riemann/blob/2f06db732bdf65108572e0a37212719a915b6c2c/src/riemann/transport/tcp.clj#L281-L284

It looks like riemann-client currently assume mutual-TLS authentication which seem aligned with this:
https://github.com/riemann/riemann-ruby-client/blob/main/lib/riemann/client/ssl_socket.rb#L20-L21

> Events work fine using tls in nodejs.

Do you say you made unauthenticated TLS work with a nodejs client?  Can you share your riemann config?  My attempts to start an unauthenticated TLS server failed…
